### ISSUE TITLE
Produce error if backtricked symbol is not declared when using {.emit.}

### ISSUE THREAD

**Issue by endragor** [nim-lang/Nim#3791]
Right now, when putting Nim symbols inside `{.emit.}` pragma, it replaces them with actual symbols in the compiled code, but only if the symbol is defined in the current scope. Otherwise, it outputs the symbol as is, which leads to successful compilation by Nim, but failure on C/C++ compilation stage.

I think it would be better to produce error when the symbol inside backtricks is not defined.

**Comment by Araq** [comment]
Maybe but who knows what might break this. Maybe we should produce a warning first?

**Comment by endragor** [comment]
Yes, that makes sense - produce warning in upcoming releases, but turn them into errors some time before 1.0.

**Comment by metagn** [comment]
For the bracketed version of `emit`, something like this was done in #22074
